lib/python_helper_sympy.py

```diff
@@ -226,9 +275,7 @@ def evaluate(expr, locals_for_eval={}):
         locals = {**locals, **locals_for_eval[key]}
     return eval(compile(root, '<ast>', 'eval'), {'__builtins__': None}, locals)
 
-def convert_string_to_sympy(a, variables, allow_hidden=False, allow_complex=False):
-    const = _Constants()
-
+def convert_string_to_sympy_with_context(a, variables, const, allow_hidden=False, allow_complex=False):
```

Comment:
@jonatanschroeder 
Based on my testing and what I think the code does (though granted I am not very experienced with the operation of the whole combined PL stack just yet), the first function without the parameter `const` is just a default fallback that is there to ensure compatibility if that older signature for the function gets called elsewhere external to what I did to modify `pl-symbolic-input`. The second form of the function named `convert_string_to_sympy_with_context` is what actually is getting called from within `pl-symbolic-input.py`. A lot of growing pains and effort went into setting up the ability to insert the custom function names and definitions -- and it has to get reconfigured every time a new tag instance is created as opposed to the original globally static dicts that do not require any changes to its member data as the whole HTML file is parsed. I think it does this every time a new `<pl-symbolic-input ...>` tag is created by re-initializing the python interpreter (again, I surmise, but am not 100% sure that this happens each tag invocation). Basically, I didn't want to break anything that relies on the way the sources before my branch used the functions in `python_helper_sympy.py`. 

I need another RedBull for this morning. Can we settle on being happy for now that it works as expected?

Reply:
Note that, if in Python you create an argument with a default value, you can skip that argument in the function call and have the same result. So 
```suggestion
def convert_string_to_sympy(a, variables, const = None, allow_hidden=False, allow_complex=False):
    if const is None:
        const = _Constants()
```
would give you the same result, without the need to provide a new function. In other words, you can call `pl.convert_string_to_sympy(a, vars)` or `pl.convert_string_to_sympy(a, vars, c)` (or even `pl.convert_string_to_sympy(a, vars, const=c)`) and all of them would be accepted as valid.

Reply:
I think in this case, the new parameter (`const`) should go at the end, so that calls that previously had relied on the third parameter being `allow_hidden` will not break.

Reply:
This is still pending.

Reply:
@maxieds If you agree, can the new param be moved to the end?

Reply:
Just to resume the conversation about this point here, the idea is not just to move the param to the end but to combine these two versions of the functions. I agree we want to preserve backwards compatibility, but essentially the `_with_context` version you've written _is_ the old version modified (it has the guts), whereas the new `convert_string_to_sympy` is just a wrapper. So we're already messing with existing implementation call stacks here, etc.